packages/jest-runtime/src/index.ts

```diff
@@ -1172,8 +1187,28 @@ class Runtime {
       });
     };
 
+    const runtime = this;
+
     // should we implement the class ourselves?
-    class Module extends nativeModule.Module {}
+    class Module extends nativeModule.Module {
+      _compile(content: string, filename: string) {
+        if (typeof content !== 'string') {
+          throw new TypeError('Module#_compile must receive string content');
+        }
+
+        if (typeof filename !== 'string') {
+          throw new TypeError('Module#_compile must receive string filename');
+        }
+
+        return runtime._execModule(
+          this,
```

Comment:
`jest-runtime`'s internal `InitialModule` is not `instanceof Module`, I believe? It probably should, but if not (and it's hard to make it so) I think constructing a new object here rather than passing `this` makes sense

Reply:
You're suggestion we make a fake `Module` instead, and provide that to the application code instead of leveraging the existing `jest-runtime`-specific instance? That seems reasonable to me, and would fix the issues around `filename` and `filenameOverride` that @jeysal identified.

Reply:
The current `jest-runtime` one should be `Module` ideally, yeah, then reused here. I think? 😅